addon/components/frost-table-row.js

```diff
@@ -65,6 +65,11 @@ export default Component.extend({
 
   // == DOM Events ============================================================
 
+  click (event) {
+    event.stopPropagation()
+    this.$().find('.frost-table-row-selection').trigger('click')
```

Comment:
So frost-table-row when you enable selection adds an frost-table-row-selection to the row. It is within the click event handler of frost-table-row-selection that the onSelect callback is call, thought it would be simplist to just trigger this event